`DialogPrimitive` with `Modal` and `Flyout` (non-breaking) - Documentation (#2213)
hashicorp/design-system: website/docs/components/flyout/partials/version-history/4.6.0.md

## 4.6.0

### Deprecated

> In this version we have replaced the existing Flyout sub-components with the equivalent sub-components from the [DialogPrimitive](/utilities/dialog-primitive) component (shared with the [Modal](/components/modal) component). They're not used anymore and for this reasons we consider them deprecated, and they'll be removed in our next major release.

Reviewer: ```suggestion
In this version, the Flyout sub-components have been deprecated and replaced with the equivalent [DialogPrimitive](/utilities/dialog-primitive) sub-components (shared also with the [Modal](/components/modal) component). The deprecated Flyout sub-components will be removed in the next major release.
```

To do the migration, if you use the `<Hds::Flyout::***>` invocation:

| Deprecated:           | Replace with:                  |
|-----------------------|--------------------------------|
| `Flyout::Header`      | `DialogPrimitive::Header`      |
| `Flyout::Description` | `DialogPrimitive::Description` |
| `Flyout::Body`        | `DialogPrimitive::Body`        |
| `Flyout::Footer`      | `DialogPrimitive::Footer`      |

If instead you use the `(component hds/flyout/***)`:

| Deprecated:              | Replace with:                     |
|--------------------------|-----------------------------------|
| `hds/flyout/header`      | `hds/dialogprimitive/header`      |
| `hds/flyout/description` | `hds/dialogprimitive/description` |
| `hds/flyout/body`        | `hds/dialogprimitive/body`        |
| `hds/flyout/footer`      | `hds/dialogprimitive/footer`      |

> In the refactoring we have been able to maintain the existing CSS class names, to avoid breaking existing tests and CSS overrides in the consumers' codebases.

Reviewer: ```suggestion
In the refactoring, we have been able to maintain existing CSS class names to avoid breaking current tests and CSS overrides in consumers' codebases.
```
